### DISCUSSION TITLE
Need help with BOOT config (Failed to write to controller)

### DISCUSSION THREAD

**Discussion by SnakeFizer** [devcompl/TTController#206] [Configuration] [answered]
Hello.
I am in need of assistance or a simple advice.

Wanted to test if **TTController** works for me and hopefully fixes **TTRGBPlus** SW conflicting with my GPU and discrete audio drivers (and it does! 👍 ).
I have created a simple config for testing purposes and everything is working great, except for two things.
1) After shutting down TTController service, it applies "Shutdown" profile from config, but fans get reset after few seconds to their default vendor state (freaking rainbow)... I guess the only way to fix this is to write a "BOOT" config to the controllers, is this correct?

2) After deleting the line from "_TTController.Service.exe.Config_": (simply changing it to "False" is leading to duplicating <bool> as "False,True" values after next launch)
`    <add key="boot-profile-saved" value="True" />`
Log\Console shows errors:
```
|INFO|TTController.Service.Managers.DeviceManager|Initialized "Riing Quad" controller [9802, 8800], version: "3.0.3"
|INFO|TTController.Service.Managers.SensorManager|Enabling sensor: /intelcpu/0/temperature/6
|INFO|TTController.Service.TTService|Processing profile "Top"
|INFO|TTController.Service.TTService|Processing profile "Bottom"
|INFO|TTController.Service.TTService|Applying computer state profile: Boot
|FATAL|TTController.Service.Hardware.HidDeviceProxy|Failed to read from [9802, 8800]!
|FATAL|TTController.Service.Hardware.HidDeviceProxy|Failed to write to [9802, 8800]!
```
And after that, controller is no longer detected until reboot:
![Screenshot 2023-01-30 171423](https://user-images.githubusercontent.com/57303232/215474181-8b468567-0dbe-46a1-962a-11ec51226151.png)

All I need is a way to setup two general profiles (Silent\Turbo) for my system and to be able to switch between them (separate configs with some batch magic?). And I don't even need RGB lighting, but I do have several spare "Riing Plus 14 RGB" 3-fan-packs...

[My debug_report.txt](https://github.com/MoshiMoshi0/TTController/files/10535477/debug_report.txt)
[My config.json in config.zip](https://github.com/MoshiMoshi0/TTController/files/10535661/config.zip)

Logs:
Everything is working OK - [normal_operation.log](https://github.com/MoshiMoshi0/TTController/files/10535809/normal_operation.log)
Failed to write boot profile - [failed.log](https://github.com/MoshiMoshi0/TTController/files/10535811/failed.log)

______________________
**P.S. - @MoshiMoshi0, a huge thank you for this alternative!** 
Might not be easy to setup for most people, but definitely a life-saviour 🥇  :) 
_Never thought I (w)could hate any kind of brand like TT that much..._

**Comment by MoshiMoshi0** [comment, marked as answer]
IIRC some/all? Quad controllers dont allow to save default speed/colors (called boot profile in TTController) and just default to rainbow colors when they dont receive data for few seconds. Thats why you need to use `IgnoreColorCache`.

So just remove boot/shutdown profiles as they are useless with Quad controllers.

**Reply by SnakeFizer** [reply]
Thank you for your prompt reply and confirmation.
I think I'm going to stick with TTController anyway, since it's lighter and does not conflict with any other HW\Drivers in my system. 

So **Thank you**, once again! Wishing you all the best ;)
